### ISSUE TITLE
Not fixed: ClassCircularityError on Tomcat 6 deployment

### ISSUE THREAD

**Issue by rhoposit** [play2war/play2-war-plugin#138]
Reference  Issue #80 - Not Fixed

play2war 0.9 RC2
Tomcat 6 (servletVersion 2.5)
play 2.1.0
scala 2.10
java 1.6.0_41
sbt.version=0.12.2

After creating WAR file, when deploying on Tomcat 6, I get the following error: 

ERROR:
Mar 11, 2013 10:09:49 PM org.apache.catalina.loader.WebappClassLoader loadClass
FINE:   Loading class from local repository
Mar 11, 2013 10:09:49 PM org.apache.catalina.loader.WebappClassLoader loadClass
FINE: loadClass(java.lang.Void, false)
java.lang.reflect.InvocationTargetException
    at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
    at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:57)
    at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
    at java.lang.reflect.Method.invoke(Method.java:616)
    at org.apache.catalina.startup.Bootstrap.start(Bootstrap.java:289)
    at org.apache.catalina.startup.Bootstrap.main(Bootstrap.java:414)
Caused by: java.lang.ClassCircularityError: java/io/PrintWriter
    at net.bull.javamelody.LoggingHandler.addErrorLogToCounter(LoggingHandler.java:69)
    at net.bull.javamelody.LoggingHandler.publish(LoggingHandler.java:99)
    at java.util.logging.Logger.log(Logger.java:530)
    at java.util.logging.Logger.doLog(Logger.java:552)
    at java.util.logging.Logger.logp(Logger.java:752)
    at org.apache.juli.logging.DirectJDKLog.log(DirectJDKLog.java:167)
    at org.apache.juli.logging.DirectJDKLog.error(DirectJDKLog.java:135)
    at org.apache.catalina.startup.HostConfig.deployWAR(HostConfig.java:970)
    at org.apache.catalina.startup.HostConfig.deployWARs(HostConfig.java:778)
    at org.apache.catalina.startup.HostConfig.deployApps(HostConfig.java:504)
    at org.apache.catalina.startup.HostConfig.start(HostConfig.java:1317)
    at org.apache.catalina.startup.HostConfig.lifecycleEvent(HostConfig.java:324)
    at org.apache.catalina.util.LifecycleSupport.fireLifecycleEvent(LifecycleSupport.java:142)
    at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1065)
    at org.apache.catalina.core.StandardHost.start(StandardHost.java:840)
    at org.apache.catalina.core.ContainerBase.start(ContainerBase.java:1057)
    at org.apache.catalina.core.StandardEngine.start(StandardEngine.java:463)
    at org.apache.catalina.core.StandardService.start(StandardService.java:525)
    at org.apache.catalina.core.StandardServer.start(StandardServer.java:754)
    at org.apache.catalina.startup.Catalina.start(Catalina.java:595)
    ... 6 more

**Comment by dlecan** [comment]
Thanks for reporting this issue.

Can you add more informations:
- Why is there a Javamelody handler in your Play application ?
- Can you copy/paste a dump of the content of `YOUR_WAR_FILE.war!/WEB-INF/lib` ?

**Comment by rhoposit** [comment]
Hello, this may be solved - at least I have a working state at the moment. The my-app.war/WEB-INF/lib directory was/is empty. 

Solution: here is what I found on the site which seems to have solved the issue for me:
https://github.com/dlecan/play2-war-plugin/wiki/ConfigurationLoggingJBoss7

Adding the following dependency seems to solve my issue:
"com.github.play2war.ext" %% "redirect-playlogger" % "1.0.1"

The javamelody jar was not in my Play application, however there are several other webapps that do use it in my webapps directory. I had not had any trouble with deploying my war file using Play2War until I began to customize my logging information.
